### ISSUE TITLE
incorrect labels in export_embeddings.py

### ISSUE THREAD

**Issue by pavelkrolevets** [davidsandberg/facenet#653]
Hi, thanks for a great work!

But it seems like the out label order in label_strings is messed up. I tried to cluster lfw but found that it gives me no results. I would be appreciated if you check it.

**Comment by cjekel** [comment]
Can you verify that your directory is set up in the correct openface format?

Your directory setup of my_database should look like:
```
my_database
└───a_person
│   │   image00.jpg
│   │   image01.jpg
│   
└───b_person
│   │   image00.jpg
│   
└───c_person
│   │   image00.jpg
│   │   image01.jpg
│   │   image02.jpg
│   │   image03.jpg
```

Where your label_strings will return  label_strings = ['a_person', 'a_person', 'b_person', 'c_person','c_person','c_person','c_person']

as posted in https://github.com/davidsandberg/facenet/issues/496

---

I just found a bug where label_strings was not the same length as labels. It's been awhile since I've worked on this script, but I don't think this was my intention. 

I've now made it so label_strings.npy is the same size as labels.npy.

Can you verify that this script fixes your problem? If so then I'll send a PR.
https://github.com/cjekel/facenet/blob/master/contributed/export_embeddings.py

**Comment by pavelkrolevets** [comment]
Hi,
Thanks for a quick answer. Yes, my directory is set up in the correct openface format. I use LFW dataset.

The problem I think at string #61:
label_strings = [name for name in os.listdir(os.path.expanduser(args.data_dir)) if os.path.isdir(os.path.join(os.path.expanduser(args.data_dir), name))]

If you save images paths (image_list) you can see that labes are different from the paths.
For example for LFW dataset: 
labels[14] => 10
label_strings[14] => Alfonso_Cuaron
image_list[14] => /home/pavel/PycharmProjects/facenet/data/lfw/Abbas_Kiarostami/Abbas_Kiarostami_0001.jpg

So, the label is wrong. I just used full img paths as labels to make it working.

**Comment by cjekel** [comment]
Does this commit fix your problem https://github.com/cjekel/facenet/commit/189bfa1e81917dcf6e30f027853fcf49e40d29d0

Because it should now be 
label_strings[14] => Abbas_Kiarostami
image_list[14] => /home/pavel/PycharmProjects/facenet/data/lfw/Abbas_Kiarostami/Abbas_Kiarostami_0001.jpg

**Comment by pavelkrolevets** [comment]
Hi, 
I checked your [commit](https://github.com/cjekel/facenet/commit/189bfa1e81917dcf6e30f027853fcf49e40d29d0) before writing previous post. It doesnt solve the problem. The problem is still in label_strings because you form it from walking folder path, which returns folder names in a different order than it is in the train_set.

I [fixed](fc8b963105fd0795f478ebd3dd028975ae20a8a4) it using paths from train_set.

Just wanted to pint out to the bag. Cheers!

**Comment by cjekel** [comment]
Sorry I just wanted to be explicitly clear whether that commit fixed your issue. 

I'm unable to reproduce the issue. The paths are always in the same order on my system? or my directory isn't giving me issues... 

However I did notice this classes.sort() in https://github.com/davidsandberg/facenet/blob/28d3bf2fa7254037229035cac398632a5ef6fc24/src/facenet.py#L331
 which is not used in the export_embedding script. Perhaps the sorting causes the issues.

I've included a new commit which uses the exact same method as get_dataset() to fetch label_strings. Does this fix the issue? https://github.com/cjekel/facenet/commit/cebc2555904ceb5f4c1ad393bcce75c088282cb2
